server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -841,6 +846,9 @@ public void onResponse(Void aVoid) {
                     Strings.format("successfully sent shard snapshot state [%s] update to the master node", status.state())
                 );
                 logger.trace("[{}][{}] updated snapshot state to [{}]", shardId, snapshot, status);
+                if (status.state().completed()) {
```

Comment:
Do you know how State.ABORTED (completed==false) works with status updates? If those are impossible to send, then maybe an assert?

Reply:
We don't send `ABORTED` indeed but we do send `PAUSED` which isn't complete and wouldn't be valid to check here (we could unpause before the check runs)

Reply:
Added an assert in 0fe6ae070d44de8b4f81533eabe497bc6005c8ad, does that clarify?

Reply:
Ohh, this is `SnapshotsInProgress.ShardState`, not `SnapshotsInProgress.State`. Oops.

Then this consistency checker won't cover the shutdown situation, with PAUSED, where we see the mystery bug. Too bad. But we should hopefully surface something in regular cluster activity sans shutdown.